services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/domain/CreateHankeRequest.kt

```diff
@@ -0,0 +1,63 @@
+package fi.hel.haitaton.hanke.domain
+
+import fi.hel.haitaton.hanke.SuunnitteluVaihe
+import fi.hel.haitaton.hanke.TyomaaTyyppi
+import fi.hel.haitaton.hanke.Vaihe
+import io.swagger.v3.oas.annotations.media.Schema
+
+data class CreateHankeRequest(
+    @field:Schema(
+        description = "Name of the project, must not be blank.",
+        maxLength = 100,
+    )
+    override val nimi: String,
+    @field:Schema(
+        description = "Shared Public Utility Site (Yhteinen kunnallistekninen työmaa). Optional.",
+    )
+    val onYKTHanke: Boolean? = null,
+    @field:Schema(
+        description =
+            "Description of the project and the work done during it. Required for the project to be published.",
+    )
+    val kuvaus: String? = null,
+    @field:Schema(
+        description = "Current stage of the project. Required for the hanke to be published.",
+    )
+    override val vaihe: Vaihe? = null,
+    @field:Schema(
+        description = "Current planning stage, must be defined if vaihe = SUUNNITTELU",
+    )
+    override val suunnitteluVaihe: SuunnitteluVaihe? = null,
+    @field:Schema(
+        description =
+            "Project owners, contact information. At least one is required for the hanke to be published.",
+    )
+    override val omistajat: List<HankeYhteystieto>? = null,
```

Comment:
Would these require a request object too? It contains id, timestamps, etc.

Reply:
The same is true about alueet. And for those, we might also need a new Geometria class.

Reply:
I think HAI-1893 can cover these. But subtasking / different pr:s sound good.